src/Workspaces/Core/Portable/Progress/CodeAnalysisProgress.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Microsoft.CodeAnalysis.Progress;
+
+namespace Microsoft.CodeAnalysis;
+
+/// <summary>
+/// Represents the progress of an operation.  Commonly used to update a UI visible to a user when a long running
+/// operation is happening.
+/// </summary>
+public sealed class CodeAnalysisProgress
+{
+    /// <summary>
+    /// Used when bridging from an api that does not show progress to the user to an api that can update progress if
+    /// available.  This should be used sparingly.  Locations that currently do not show progress should ideally be
+    /// migrated to ones that do so that long running operations are visible to the user in a coherent fashion.
+    /// </summary>
+    internal static readonly IProgress<CodeAnalysisProgress> None = NullProgress<CodeAnalysisProgress>.Instance;
+
+    internal bool ClearValue { get; init; }
+    internal int? CompleteItemValue { get; init; }
```

Comment:
It seems a little odd to me that this class doesn't expose getters for these. I could see cases where api user has a method that creates a CodeAnalysisProgress and the calling method wants to inspect the result before reporting it.

Reply:
but the progress guy is created directly from the data the caller passes in.  what case would they have to do that, then want to query the very data they passed into this?

I really want this to act a sink.  you have your data, you package it into the messag,e you send the message to the progress sink.

on the feature side, it's very unidirectional in that fashion.

Reply:
I probably don't have a great feel yet for usage of this, but I was curious about the case where maybe somebody calls into a method that returns one or many of these CodeAnalysisProgress items and is responsible for reporting those updates. Perhaps it would wanted to batch/filter/merge the changes.